dnf-docker-test/features/plugin-api-1.feature

```diff
@@ -0,0 +1,39 @@
+Feature: DNF test (plugin api)
+
+@bz1626093
+Scenario: Plugin can edit http headers
+  Given http repository "base" with packages
+    | Package | Tag       | Value  |
+    | TestA   |           |        |
+  When I save rpmdb
+  And I enable repository "base"
+  Given a file "/test/plugins/test.py" with 
+  """
+  import dnf
+  import dnf.cli
+  import dnf.exceptions
+  import dnf.rpm.transaction
+
+  class Test(dnf.Plugin): 
+
+    name = "test" 
+    def __init__(self, base, cli):
+        super(Test, self).__init__(base, cli)
+
+    def config(self): 
+        for repoid, repo in self.base.repos.items(): 
+            repo.set_http_headers([ 
+                "custom_user_key20190218: custom_user_value", 
+            ])
+  """
+  Given a file "/etc/dnf/dnf.conf" with 
+  """
+  [main]
+  pluginpath=/test/plugins
+  """
+  And I successfully run "dnf clean all"
+  Then I sniff packets expecting "custom_user_key20190218: custom_user_value"
+  Given a file "/etc/dnf/dnf.conf" with 
```

Comment:
What is the purpose of this Given? Is this a clean up step (which is not necessary)? Or is the test somehow incomplete?

Reply:
It was supposed to be clean up, I originally had it in different feature with other tests. You are right, it is not necessary.